File: prism-sdk/crypto/jvm/src/main/java/io/iohk/atala/prism/crypto/japi/ExtendedKey.java
package io.iohk.atala.prism.crypto.japi;

public interface ExtendedKey {
    /**
     * Derivation path used to obtain such key */
    DerivationPath getPath();

    /**
     * Public key for this extended key */
    ECPublicKey getPublic();

    /**
     * Private key for this extended key */
    ECPrivateKey getPrivate();

    /**
     * KeyPair for this extended key */
    ECKeyPair getKeyPair();

    /**
     * Generates child extended key for given index */
    ExtendedKey derive(DerivationAxis axis);
}

File: prism-kotlin-sdk/crypto/src/commonMain/kotlin/io/iohk/atala/prism/kotlin/crypto/derivation/ExtendedKey.kt
package io.iohk.atala.prism.kotlin.crypto.derivation

import io.iohk.atala.prism.kotlin.crypto.keys.ECKeyPair
import io.iohk.atala.prism.kotlin.crypto.keys.ECPrivateKey
import io.iohk.atala.prism.kotlin.crypto.keys.ECPublicKey

interface ExtendedKey {
    /** Derivation path used to obtain such key */
    fun path(): DerivationPath

    /** Public key for this extended key */
    fun publicKey(): ECPublicKey

    /** Private key for this extended key */
    fun privateKey(): ECPrivateKey

    /** KeyPair for this extended key */
    fun keyPair(): ECKeyPair

    /** Generates child extended key for given index */
    fun derive(axis: DerivationAxis): ExtendedKey
}
